api/specs/storage/scripts/storage.py

```diff
@@ -343,9 +344,11 @@ async def delete_folders_of_project(
     response_model=Envelope[FileMetaDataGet],
     tags=TAGS_SIMCORE_S3,
     summary="search for files starting with",
-    operation_id="search_files_starting_with",
+    operation_id="search_files",
 )
-async def search_files_starting_with(user_id: UserID, startswith: str = ""):
+async def search_files(
+    user_id: UserID, startswith: str = "", sha256_checksum: SHA256Str | None = None
```

Comment:
Let's have a chat about this one. Here are my issues:
- the searching is a bit of a black box, it goes through the `file_metada_data` and if the entry is a directory it will search the via S3. With files stored in S3 a checksum as you defined cannot be provided easily nor we can store it
- Why is it not possible to use the previous search method? What is your use case here? Can we maybe find an alternative?